### ISSUE TITLE
`test.AbsolutePositionForObject` and custom widgets

### ISSUE THREAD

**Issue by AlbinoGeek** [fyne-io/fyne#1841]
### Describe the bug:

`test.AbsolutePositionForObject` always returns `0,0` for a custom widget

### To Reproduce:

Steps to reproduce the behaviour:

1. Extend a widget (such as *cell does below)
1. Write a test that renders the widget in the UI
1. Attempt to get the `AbsolutePositionForObject`

#### `main_test.go`

```go
func TestUICells(t *testing.T) {
	var (
		a        = test.NewApp()
		board, w = start(a)
	)
	a.Run()

	assert.NoError(t, board.load(wikipedia), "failed loading valid classic sudoku")
	test.AssertImageMatches(t, "start.png", w.Canvas().Capture())

	drv := test.NewDriver()
	pos := drv.AbsolutePositionForObject(board.cells[80])
	test.MoveMouse(w.Canvas(), pos)
	test.TapCanvas(w.Canvas(), pos)
	t.Logf("%#v", pos) // fyne.Position{X:0, Y:0}
	t.Fail()           // forces log to show
}
```

#### `cell.go`

```go
var _ fyne.Widget = (*cell)(nil)
var _ fyne.Disableable = (*cell)(nil)
var _ desktop.Hoverable = (*cell)(nil)
var _ desktop.Mouseable = (*cell)(nil)
var _ mobile.Touchable = (*cell)(nil)

type cell struct {
	widget.BaseWidget `json:"-"`
	// ...
}

func newCell(id int) *cell {
	c := &cell{}
	c.ExtendBaseWidget(c)

	return c
}
```

### Device (please complete the following information):

- **OS:** Fedora Workstation
- **Version:** `33`
- **Go version:** `1.15.6 linux/amd64`
- **Fyne version:** `2.0.0-rc5`

**Comment by andydotxyz** [comment]
I don’t think this relates to custom widgets, this is due to your element not being on the canvas. You can’t get the absolute position for something that is not placed in the object tree for the canvas.

**Comment by AlbinoGeek** [comment]
It is on the Canvas, as an image test before confirms it and many other cells are rendered:

![start](https://user-images.githubusercontent.com/1910461/105331872-d0b2f100-5b88-11eb-8462-4afd9593581b.png)

**Comment by andydotxyz** [comment]
It turns out this is not related to custom widgets, but how many windows are loaded.
The current implementation is quite simple and only works for elements added to the most recently opened window

**Comment by andydotxyz** [comment]
If you could share the `start()` function we could explore this more, because how it creates the window may be part of why it's not working for you.

Also `test.Run()` is a no-op. Test apps don't really run, they just manage state and render when requested.

**Comment by AlbinoGeek** [comment]
I only have one window, the `start` function is as follows, this was practically the re-used code from `main()` that was pulled out for testing sake:

```go
func start(a fyne.App) (*board, fyne.Window) {
	w := a.NewWindow("Number Place")
	b := newBoard(3, 3, 3, 3)

	uiInit(b, w)

	w.Show()

	return b, w
}
```
